components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/oauth/OAuthAuthenticator.java

```diff
@@ -168,7 +179,46 @@ public AuthenticationResponse authenticate(MessageContext synCtx) {
         }
         String authenticationScheme;
         try {
-            authenticationScheme = getAPIKeyValidator().getResourceAuthenticationScheme(synCtx);
+            //Initial guess of a JWT token using the presence of a DOT.
+            if (apiKey.contains(APIConstants.DOT)) {
+                try {
+                    JSONObject decodedHeader = new JSONObject(new String(Base64.getUrlDecoder()
+                            .decode(apiKey.split("\\.")[0])));
+                    // Check if the decoded header contains type as 'JWT'.
+                    if (APIConstants.JWT.equals(decodedHeader.getString(APIConstants.JwtTokenConstants.TOKEN_TYPE))) {
+                        isJwtToken = true;
+                        if (StringUtils.countMatches(apiKey, APIConstants.DOT) != 2) {
+                            log.debug("Invalid JWT token. The expected token format is <header.payload.signature>");
+                            throw new APISecurityException(APISecurityConstants.API_AUTH_INVALID_CREDENTIALS,
+                                    "Invalid JWT token");
```

Comment:
Cant we state the same debug message here as well?
It provides a hint.

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/9250